sumpy/e2p.py

```diff
@@ -81,15 +84,18 @@ def __init__(self, ctx, expansion, kernels,
     def default_name(self):
         pass
 
+    @memoize_method
+    def get_cached_loopy_knl_and_optimizations(self):
+        return self.expansion.get_loopy_evaluator(self.kernels)
```

Comment:
It seems that this would simply fail if E2P were used for M2P, right?

Reply:
No, there's an implementation for M2P as well. (Just not super optimized)